pandas/io/stata.py

```diff
@@ -1848,13 +1882,15 @@ def _open_file_binary_write(fname):
     own : bool
         True if the file was created, otherwise False
     """
-    if hasattr(fname, "write"):
-        # if 'b' not in fname.mode:
+    if isinstance(fname, IOBase):
         return fname, False
-    return open(fname, "wb"), True
+    elif isinstance(fname, (str, Path)):
+        return open(fname, "wb"), True
+    else:
+        raise ValueError("fname must be a binary file, buffer or path-like.")
```

Comment:
would TypeError be more applicable?

Reply:
Yes. I think this line may be unhittable.